dvc/tree/dropbox.py

```diff
@@ -0,0 +1,466 @@
+import datetime
+import json
+import logging
+import os
+import threading
+from abc import ABCMeta, abstractmethod
+from collections import namedtuple
+from contextlib import closing
+
+from funcy import cached_property, retry, wrap_prop
+
+from dvc.exceptions import DvcException
+from dvc.hash_info import HashInfo
+from dvc.path_info import CloudURLInfo
+from dvc.progress import Tqdm
+from dvc.scheme import Schemes
+from dvc.utils import format_link
+
+from .base import BaseTree
+
+logger = logging.getLogger(__name__)
+
+
+def _dropbox_retry(func):
+    def should_retry(exc):
+        import dropbox
+
+        if not isinstance(
+            exc,
+            (
+                dropbox.exceptions.InternalServerError,
+                dropbox.exceptions.RateLimitError,
+            ),
+        ):
+            return False
+
+        logger.debug(f"Retrying Dropbox API call, error: {exc}.")
+
+        return True
+
+    # 16 tries, start at 0.5s, multiply by golden ratio, cap at 20s
+    return retry(
+        16,
+        timeout=lambda a: min(0.5 * 1.618 ** a, 20),
+        filter_errors=should_retry,
+    )(func)
+
+
+class DropboxWrapper:
+    def __init__(self, client):
+        self.client = client
+
+    @_dropbox_retry
+    def files_get_temporary_link(self, *args, **kwargs):
+        return self.client.files_get_temporary_link(*args, **kwargs)
+
+    @_dropbox_retry
+    def files_get_metadata(self, *args, **kwargs):
+        return self.client.files_get_metadata(*args, **kwargs)
+
+    @_dropbox_retry
+    def files_list_folder(self, *args, **kwargs):
+        return self.client.files_list_folder(*args, **kwargs)
+
+    @_dropbox_retry
+    def files_list_folder_continue(self, *args, **kwargs):
+        return self.client.files_list_folder_continue(*args, **kwargs)
+
+    @_dropbox_retry
+    def files_delete_v2(self, *args, **kwargs):
+        return self.client.files_delete_v2(*args, **kwargs)
+
+    @_dropbox_retry
+    def files_upload(self, *args, **kwargs):
+        return self.client.files_upload(*args, **kwargs)
+
+    @_dropbox_retry
+    def files_upload_session_start(self, *args, **kwargs):
+        return self.client.files_upload_session_start(*args, **kwargs)
+
+    @_dropbox_retry
+    def files_upload_session_append_v2(self, *args, **kwargs):
+        return self.client.files_upload_session_append_v2(*args, **kwargs)
+
+    @_dropbox_retry
+    def files_upload_session_finish(self, *args, **kwargs):
+        return self.client.files_upload_session_finish(*args, **kwargs)
+
+    @_dropbox_retry
+    def files_download(self, *args, **kwargs):
+        return self.client.files_download(*args, **kwargs)
+
+
+ACCESS_TOKEN = "DROPBOX_ACCESS_TOKEN"
```

Comment:
It looks like rclone uses a different way of authenticating? It gets ` 

`{"access_token":".***","token_type":"bearer","expiry":"0001-01-01T00:00:00Z"}`

I don't see a refresh token, and expiry is not set. Do you know something about this?

Reply:
They may be using legacy auth, which will be dropped soon by Dropbox.